components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/util/AuthenticatedUserBuilder.java

```diff
@@ -49,41 +50,67 @@ public class AuthenticatedUserBuilder {
 
     private static final Log LOG = LogFactory.getLog(AuthenticatedUserBuilder.class);
     private AuthenticatedUser authenticatedUser;
-    private final AuthenticatedUserData user;
+    private final AuthenticatedUserData userFromResponse;
     private final AuthenticationContext context;
     private final AuthenticatorAdapterConstants.UserType userType;
+    private String usernameFromResponse;
     
     public AuthenticatedUserBuilder(AuthenticatedUserData user, AuthenticationContext context) {
 
-        this.user = user;
+        this.userFromResponse = user;
         this.context = context;
         userType = resolveIdpType();
     }
 
-    private void validateUserData() throws ActionExecutionResponseProcessorException {
-
-        if (StringUtils.isBlank(user.getUser().getId())) {
-            throw new ActionExecutionResponseProcessorException("User Id is not found in the authenticated user data.");
-        }
-        if (AuthenticatorAdapterConstants.UserType.LOCAL.equals(userType) && user.getUser().getUserStore() == null) {
-            throw new ActionExecutionResponseProcessorException("User store domain is not found in the authenticated " +
-                    "user data for the local user.");
+    /**
+     * This method is responsible for building the authenticated user object from the authenticated user data.
+     *
+     * @return AuthenticatedUser object.
+     * @throws ActionExecutionResponseProcessorException If any error occurred when building the authenticated
+     * user object.
+     */
+    public AuthenticatedUser buildAuthenticateduser() throws ActionExecutionResponseProcessorException {
+
+        if ((AuthenticatorAdapterConstants.UserType.LOCAL.equals(userType))) {
+            return buildLocalAuthenticatedUserFromResponse();
         }
+        return buildFederatedAuthenticatedUserFromResponse();
     }
 
-    public AuthenticatedUser buildAuthenticateduser()
+    private AuthenticatedUser buildFederatedAuthenticatedUserFromResponse()
             throws ActionExecutionResponseProcessorException {
 
-        validateUserData();
-
-        if (AuthenticatorAdapterConstants.UserType.FEDERATED.equals(userType)) {
-            resolveFederatedUser();
-        } else {
-            resolveLocalUser();
-        }
+        String userId = resolveUserIdFromResponse();
+        authenticatedUser = AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(userId);
+        Map<ClaimMapping, String> attributeMap = resolveUserNameAndClaimsFromResponse();
+        // Set the user ID to the external ID claim for federated authenticators.
+        attributeMap.put(buildClaimMapping(EXTERNAL_ID_CLAIM), userId);
+        authenticatedUser.setUserAttributes(attributeMap);
+        resolveUsernameForFederatedUser();
```

Comment:
Why do we have two methods to resolve username. line 85,  and line 89

Reply:
Renamed `resolveUsernameForFederatedUser()` to `setUsernameForFederatedUser()`